mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs

```diff
@@ -49,22 +49,30 @@ internal static void InstallNativeHelper (INativeCertificateHelper helper)
 				Interlocked.CompareExchange (ref nativeHelper, helper, null);
 		}
 
-#if !MOBILE
-		[SecurityPermission (SecurityAction.Demand, UnmanagedCode = true)]
-#endif
 		public static X509CertificateImpl InitFromHandle (IntPtr handle)
 		{
 #if MONO_FEATURE_APPLETLS
-			var variable = Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER");
-			bool useAppleTls = string.IsNullOrEmpty (variable) || variable == "default" || variable == "apple"; // On Platform.IsMacOS default is AppleTlsProvider
-			if (System.Environment.IsMacOS && useAppleTls)
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
 				return InitFromHandleApple (handle);
 #endif
-#if !MOBILE
+#if !ONLY_APPLETLS && !MOBILE
 			InitFromHandleCore (handle);
 #endif
 			throw new NotSupportedException ();
 		}
+		
+		static X509CertificateImpl Import (byte[] rawData)
+		{
+#if MONO_FEATURE_APPLETLS
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
+				return ImportApple (rawData);
+#endif
+#if !ONLY_APPLETLS && !MOBILE
```

Comment:
Yeah I guess you are right in this case. The InitFromHandle has a version that threw 

https://github.com/mono/mono/blob/master/mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs#L85

but the Import (byte) had one of two with no throw:

https://github.com/mono/mono/blob/master/mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs#L180

I was making them a little too much the same apparently.